### PR TITLE
Fix for Locale bug (for example: Turkish locale "İ" character bug)

### PR DESCRIPTION
**Pull request by paledega**
InputMethodService class need to set manual locale.

### CODE REVIEW

**Comment by Quiec** [comment]
You are right bro.

**Comment by archi-doctor** [comment]
Yes. When I am using Turkish Q, i and ı characters are on different buttons. But their uppercases are wrong. Corrects are İ and I but on anysoftkeyboard, these two buttons are same in uppercase mode. I and I. We cant write İ character. Paledeka knoqs technical details

**Comment by paledega** [comment]
IntupMethodService class not use android default languages. So I force set locale in application.

I am developer of droidtr keyboard. (Turkish keyboard) We had this problem. We fix this bug with my sollution.

**Comment by paledega** [comment]
And keyboard language packs ask tag not working.

**Comment by archi-doctor** [comment]
You could come to Telegram to communicate with we for solvation. t.me/archi_dagac is me. t.me/sulincix is paledega. and our group is t.me/DroidTR

**Comment by menny** [comment]
`InputMethodService class not use android default languages. So I force set locale in application.`
this is terrible! Are you sure about this?

**Comment by paledega** [comment]
I am totally sure. If you dont believe You should test.

File: app/src/main/java/com/anysoftkeyboard/ime/AnySoftKeyboardBase.java [hunk 1]
```diff
@@ -188,9 +188,24 @@ protected void showOptionsDialogWithData(CharSequence title, @DrawableRes int ic
...
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N){
+            return context.getResources().getConfiguration().getLocales().get(0);
+        } else{
+            //noinspection deprecation
+            return context.getResources().getConfiguration().locale;
+        }
+    }
     @Override
     public View onCreateInputView() {
+        //Fix for Locale bug (for example: Turkish locale İ character bug)
```
[the remark this answers is not in the document]
Comment (menny) [reply app/src/main/java/com/anysoftkeyboard/ime/AnySoftKeyboardBase.java]: And, can you see how to integrate this with https://github.com/AnySoftKeyboard/AnySoftKeyboard/blob/master/app/src/main/java/com/anysoftkeyboard/utils/LocaleTools.java

**Comment by menny** [comment]
closing in favor of https://github.com/AnySoftKeyboard/AnySoftKeyboard/pull/1619

**Pull request closed without merging**